Geometry/HGCalMapping/plugins/alpaka/HGCalMappingCellESProducer.cc

```diff
@@ -41,9 +43,41 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
         desc.add<std::vector<std::string> >("filelist", std::vector<std::string>({}))
             ->setComment("list of files with the readout cells of each module");
         desc.add<edm::ESInputTag>("cellindexer", edm::ESInputTag(""))->setComment("Dense cell index tool");
+        desc.add<std::string>("offsetfile", std::string({}))
+            ->setComment("file containing the offsets between calibration and surrounding cells");
         descriptions.addWithDefaultLabel(desc);
       }
 
+      std::map<std::tuple<std::string, int, int>, int> makeOffsetMap(std::string input_offsetfile) {
+        std::map<std::tuple<std::string, int, int>, int> offsetMap;
+        auto offsetfile = edm::FileInPath(input_offsetfile).fullPath();
+        std::ifstream stream_offsetfile(offsetfile);
+        std::string line;
+        // Skip the first line (description of column content)
+        std::getline(stream_offsetfile, line);
+        while (std::getline(stream_offsetfile, line)) {
+          std::istringstream iss(line);
+          std::string r_typecode_str, r_chip_str, r_half_str, offset_str;
+          if (!(iss >> r_typecode_str >> r_chip_str >> r_half_str >> offset_str)) {
+            std::cout << "Error reading offset file" << std::endl;
+            break;
+          }
+          try {
+            int r_chip = std::stoi(r_chip_str);
+            int r_half = std::stoi(r_half_str);
+            int offset = std::stoi(offset_str);
+            offsetMap[std::make_tuple(r_typecode_str, r_chip, r_half)] = offset;
+          } catch (const std::invalid_argument& e) {
```

Comment:
If a parsing failure is acceptable behavior in the sense that you don't want to throw an exception further (and terminate the application), please use a non-throwing parser (e.g. `std::from_chars()`) instead. Exceptions should be used only in truly exceptional situations.